### PR TITLE
fix: pass unsafePerm to lifecycle call

### LINKED ISSUE #963: `pnpm install` doesn't run postinstall and preinstall scripts when `name` property in package.json is different than project dir name
### pnpm version:
1.23.1
### Code to reproduce the issue:
package.json in project directory `test`:
```
{
  "name": "test2",
  "version": "0.0.1",
  "scripts": {
    "postinstall": "echo postinstall",
    "preinstall": "echo preinstall"
  }
}
```

### Expected behavior:
```
➤ pnpm i

> test2@0.0.1 preinstall /Users/testuser/projects/test
> echo preinstall

preinstall
Already up-to-date

> test2@0.0.1 postinstall /Users/testuser/projects/test
> echo postinstall

postinstall
```
### Actual behavior:
```
➤ pnpm i
Already up-to-date
```
### Additional information:

 - `node -v` prints: v8.7.0
 - Windows, OS X, or Linux?: OS X

### PR DESCRIPTION
**Pull request by etamponi**
This fixes https://github.com/pnpm/pnpm/issues/963

### CODE REVIEW

**Comment by zkochan** [comment]
Please add a test for this use case to

https://github.com/pnpm/supi/blob/master/test/install/lifecycleScripts.ts

**Comment by etamponi** [comment]
Here you go :) this puts us in par with npm: scripts will be run by non-root users, and root will run scripts only if --unsafe-perm is set. This also requires no additional changes to `@zkochan/lifecycle`

Do you still want to cover this with a test?

**Comment by zkochan** [comment]
Thanks!

Yeah, a test is still needed. We don't want to have regressions.

File: src/api/extendOptions.ts [hunk 1]
```diff
@@ -83,5 +83,12 @@ export default async (opts?: SupiOptions): Promise<StrictSupiOptions> => {
     extendedOpts.prefix = path.join(extendedOpts.prefix, subfolder)
   }
   extendedOpts.rawNpmConfig['registry'] = extendedOpts.registry
+  if (!extendedOpts.rawNpmConfig['unsafe-perm']) {
+    extendedOpts.rawNpmConfig['unsafe-perm'] = process.platform === 'win32' ||
```
Comment (zkochan) [inline src/api/extendOptions.ts]: I added `unsafePerm` to the options https://github.com/pnpm/types/commit/a35d2a3376b9ec211b4ffb9a602237d67eeea6ef

So just update `@pnpm/types` and use it as a regular option.

**Comment by etamponi** [comment]
Is the update available already?

Il 24 dic 2017 15:14, "Zoltan Kochan" <notifications@github.com> ha scritto:

**Comment by etamponi** [comment]
Tests added :)

**Review by zkochan — APPROVED** [review]

**Comment by etamponi** [comment]
Do you want me to consolidate this PR into a single commit?

**Comment by zkochan** [comment]
yes. That would be great.

you can rewrite this branch and it will update the CR

**Comment by etamponi** [comment]
Here you go :)

**Pull request merged**